app/src/main/java/com/android/solvit/seeker/ui/review/CreateReview.kt

```diff
@@ -0,0 +1,298 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.MutableIntState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInteropFilter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.solvit.R
+import com.android.solvit.seeker.model.provider.ListProviderViewModel
+import com.android.solvit.seeker.ui.service.ProviderItem
+import com.android.solvit.shared.model.map.Location
+import com.android.solvit.shared.model.request.ServiceRequest
+import com.android.solvit.shared.model.request.ServiceRequestStatus
+import com.android.solvit.shared.model.request.ServiceRequestViewModel
+import com.android.solvit.shared.model.review.Review
+import com.android.solvit.shared.model.review.ReviewViewModel
+import com.android.solvit.shared.ui.navigation.NavigationActions
+import com.android.solvit.shared.ui.navigation.Route
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.google.maps.android.compose.rememberMarkerState
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@Composable
+fun CreateReviewScreen(
+    reviewViewModel: ReviewViewModel,
+    serviceRequestViewModel: ServiceRequestViewModel,
+    listProviderViewModel: ListProviderViewModel,
+    navigationActions: NavigationActions
+) {
+  val requestState = serviceRequestViewModel.selectedRequest.collectAsState()
+  if (requestState.value == null || requestState.value!!.status != ServiceRequestStatus.COMPLETED) {
+    navigationActions.goBack()
+  }
+  val ratingState = remember { mutableIntStateOf(0) }
+  var comment by remember { mutableStateOf("") }
+
+  // Lock Orientation to Portrait
+  val context = LocalContext.current
+  DisposableEffect(Unit) {
+    val activity = context as? ComponentActivity
+    activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+    onDispose { activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED }
+  }
+
+  Scaffold(topBar = { TopSection(navigationActions) }) { paddingValues ->
+    Column(
+        modifier =
```

Comment:
I recommend using my `goBackButton` method for the top bar, as your current implementation does not handle double-clicking on the back button, which can unexpectedly navigate back two pages.